framework-crates/objc2-network/src/path_monitor.rs

```diff
@@ -0,0 +1,76 @@
+use core::ptr::NonNull;
+
+use crate::{generated::*, nw_path_monitor_create, InterfaceType, NWRetained};
+
+nw_object!(
+    #[doc(alias = "nw_path_monitor_t")]
+    pub struct PathMonitor;
+);
+
+impl PathMonitor {
+    pub fn new() -> NWRetained<Self> {
+        nw_path_monitor_create()
+    }
+
+    pub fn new_with_interface_type(interface_type: InterfaceType) -> NWRetained<Self> {
+        nw_path_monitor_create_with_type(interface_type)
+    }
```

Comment:
There should be a way for `header-translator` to automatically emit these as associated functions, you might need to modify the `find_fn_implementor` logic in `name_translation.rs` (it's kind of a mess tbh).

Reply:
I am not entirely familiar with the header-translation pipeline. Currently it seems that many things are handled in different places but maybe that's because I am not familiar with how it works. 

For example:

```
nw_txt_record_t nw_txt_record_copy(nw_txt_record_t txt_record);
```

is generated into

```rs

impl NWTxtRecord {
  pub fn copy(txt_record: Option<&NWTxtRecord>) -> Option<NWRetained<NWTxtRecord>> { .. }
}
```

Given that we change implementer to `NWTxtRecord`, I'd have expect the first argument be replaced with `&self` but I am not sure where this transform should happen.